### PR TITLE
FIX: delete orphan post revisions

### PR DESCRIPTION
**Pull request by lis2**
I was adding specs to ensure that post actions and uploads are removed for permanently deleted posts.

I noticed that post revisions were not permanently destroyed. I added a migration to fix old data.

### CODE REVIEW

File: lib/post_destroyer.rb [hunk 1]
```diff
@@ -291,6 +292,11 @@ def trash_public_post_actions
     end
   end
 
+  def trash_revisions
+    return unless permanent?
+    @post.revisions.map(&:destroy!)
```
Comment (CvX) [inline lib/post_destroyer.rb:297]:
```suggestion
    @post.revisions.each(&:destroy!)
```

We're not using the resulting `true`-filled array 😃

File: db/migrate/20210324043327_delete_orphan_post_revisions.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class DeleteOrphanPostRevisions < ActiveRecord::Migration[6.0]
```
Comment (CvX) [inline db/migrate/20210324043327_delete_orphan_post_revisions.rb:3]: Should this be a migration or a post_migration? 🤔

Comment (gschlager) [reply db/migrate/20210324043327_delete_orphan_post_revisions.rb:3]: The app doesn't depend on this change and it's potentially a slow SQL, so I'd move it into a post migration.

**Review by CvX — APPROVED** [review]
Looks good 👍

**Comment by CvX** [comment]
@SamSaffron would `disable_ddl_transaction!` be in any way helpful (or necessary?) in simple single-query cases like this?

**Pull request merged**